src/core/render-core.js

```diff
@@ -81,10 +81,12 @@ async function render(_opts = {}) {
       await page.emulateMedia('screen');
     }
 
-    logger.info('Setting cookies..');
-    opts.cookies.map(async (cookie) => {
-      await page.setCookie(cookie);
-    });
+    if (opts.cookies && opts.cookies.length > 0) {
```

Comment:
Hi, thanks for the efforts! This PR seems to also change the cookie logic. Could you separate the cookie change from this PR to another one and explain why it's also needed?